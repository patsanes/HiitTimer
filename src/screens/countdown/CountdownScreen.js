import React from 'react';
import { View } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import PropTypes from 'prop-types';
import { StartCountdownContainer } from './containers';
import { goToHome } from '../../navigation/actions';
import styles from './styles';

class CountdownScreen extends React.Component {
  render() {
    const { navigation } = this.props;
    const { dispatch } = navigation;
    const goHome = () => dispatch(goToHome());

    return (
      <View style={styles.container}>
        <LinearGradient
          start={{ x: 0.0, y: 0.25 }}
          end={{ x: 0.5, y: 1.0 }}
          locations={[0, 0.5]}
          colors={['#FF416C', '#FF4B2B']}
          style={styles.linearGradient}
        >
          <View style={styles.middle}>
            <StartCountdownContainer goToHome={goHome} />
          </View>
        </LinearGradient>
      </View>
    );
  }
}

CountdownScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
};

export default CountdownScreen;
