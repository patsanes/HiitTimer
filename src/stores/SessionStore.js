import { types } from 'mobx-state-tree';
import moment from 'moment';

export const SessionStore = types
  .model('SessionStore', {
    serie: types.number,
    currentSerie: types.number,
    cycle: types.number,
    currentCycle: types.number,
    training: types.number, // seconds for training
    rest: types.number, // seconds for resting
    restBetween: types.number, // seconds for resting ADD THIS LATER ON
    startCountdown: types.number,
    timeCompleteWorkout: types.number,
    isPlay: types.boolean,
    isStop: types.boolean,
    isRest: types.boolean,
    currentTime: types.number,
    inProgress: types.boolean,
    timePased: types.number,
  })
  .views(self => ({
    get endTime() {
      const startTime = '00:00:00';
      const extraSeconds = (self.rest + self.training) * self.serie * self.cycle;
      const endTime = moment(startTime, 'HH:mm:ss').add(extraSeconds, 'seconds');
      return endTime.format('H [horas] m [min y] s [seg]');
    },
  }))
  .actions(self => ({
    setInProgress() {
      self.inProgress = !self.inProgress;
    },
    setPause() {
      self.isPlay = false;
    },
    setPlay() {
      self.isPlay = true;
      self.isStop = false;
    },
    setStop() {
      if (!self.isStop) {
        self.isPlay = false;
        self.isStop = true;
        // self.resetTime();
        self.setInProgress();
      }
    },
    increaseTime() {
      self.timePased = self.timePased + 1;
    },
    saveTime() {
      self.currentTime -= 1;
    },
    resetTime(save) {
      if (save) {
        self.saveTime();
      } else {
        if (self.isRest) {
          self.currentTime = self.rest;
        } else {
          self.currentTime = self.training;
        }
      }
    },
    increaseSerie() {
      if (!self.isRest) {
        if (self.currentSerie === self.serie) {
          self.currentSerie = 0;
          self.increaseCycle();
        } else {
          self.currentSerie += 1;
        }
      }
      self.isRest = !self.isRest;
      self.resetTime(false);
    },
    increaseCycle() {
      if (self.currentCycle === self.cycle) {
        self.setPause();
        if (self.isTimeToFinish()) {
          self.finishWorkout();
        }
      } else {
        self.currentCycle += 1;
      }
    },
    isTimeToFinish() {
      if (
        self.serie == self.currentSerie &&
        self.cycle == self.currentCycle &&
        self.timePased == self.timeCompleteWorkout
      ) {
        return true;
      }
      return false;
    },
    finishWorkout() {
      self.setStop();
    },
    updateTraining(newValue) {
      self.training = Number(newValue);
    },
    updateRest(newValue) {
      self.rest = Number(newValue);
    },
    updateCycle(newValue) {
      self.cycle = Number(newValue);
    },
    updateSerie(newValue) {
      self.serie = Number(newValue);
    },
    updateRestBetween(newValue) {
      self.restBetween = Number(newValue);
    },
    updateStartCountdown(newValue) {
      self.startCountdown = Number(newValue);
    },
  }));

const initialState = {
  serie: 1,
  currentSerie: 0,
  cycle: 1,
  currentCycle: 0,
  training: 5, // seconds for training
  rest: 3, // seconds for resting
  restBetween: 3, // seconds for resting
  startCountdown: 2, // seconds for resting
  isPlay: false,
  isStop: true,
  isRest: false,
  inProgress: false,
  timePased: 0,
};
initialState.currentTime = initialState.training;
initialState.timeCompleteWorkout =
  (initialState.training + initialState.rest) * initialState.serie * initialState.cycle;

export { initialState };
