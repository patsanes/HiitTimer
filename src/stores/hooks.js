import { useContext } from 'react';
import { MobXProviderContext } from 'mobx-react';

const useStores = () => {
  return useContext(MobXProviderContext);
};

// eslint-disable-next-line import/prefer-default-export
export { useStores };
