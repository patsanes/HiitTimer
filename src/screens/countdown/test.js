import React from 'react';
import { render } from 'react-native-testing-library';
import { Provider } from 'mobx-react';

import { initialState, SessionStore } from 'HiitTimer/src/stores/SessionStore';
import CountdownScreen from './CountdownScreen';

// jest.mock('react-native-sound', () => ({
//   setCategory: () => undefined,
// }));

const navigation = {
  goBack: () => undefined,
};

const session = SessionStore.create(initialState);
describe('CountdownScreen', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <Provider session={session}>
        <CountdownScreen navigation={navigation} />
      </Provider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
