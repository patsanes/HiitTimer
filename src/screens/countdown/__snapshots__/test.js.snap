// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountdownScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <BVLinearGradient
    colors={
      Array [
        4294821936,
        4294144821,
      ]
    }
    endPoint={
      Object {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={
      Array [
        0,
        0.5,
      ]
    }
    startPoint={
      Object {
        "x": 0,
        "y": 0.25,
      }
    }
    style={
      Object {
        "borderRadius": 5,
        "flex": 1,
        "paddingLeft": 15,
        "paddingRight": 15,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 7,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "fontFamily": "League Gothic",
            "fontSize": 250,
            "fontWeight": "bold",
            "position": "absolute",
          }
        }
      >
        3
      </Text>
    </View>
  </BVLinearGradient>
</View>
`;
